src/main/java/de/tum/in/www1/artemis/service/programming/ProgrammingExerciseImportFromFileService.java

```diff
@@ -85,11 +83,29 @@ public ProgrammingExercise importProgrammingExerciseFromFile(ProgrammingExercise
         if (Boolean.TRUE.equals(programmingExerciseForImport.isStaticCodeAnalysisEnabled())) {
             staticCodeAnalysisService.createDefaultCategories(importedProgrammingExercise);
         }
+        copyEmbeddedFiles(exerciseFilePath.toAbsolutePath().getParent().resolve(FileNameUtils.getBaseName(exerciseFilePath.toString())));
         importRepositoriesFromFile(importedProgrammingExercise, importExerciseDir, oldShortName);
         importedProgrammingExercise.setCourse(course);
+        fileService.scheduleForDirectoryDeletion(importExerciseDir, 5);
         return importedProgrammingExercise;
     }
 
+    /**
+     * copy embedded files from the extracted zip file to the markdown folder, so they can be used in the problem statement
+     **/
+    private void copyEmbeddedFiles(Path importExerciseDir) throws IOException {
+        // the conversion to a string is needed because otherwise the file extension is not included
+        Predicate<Path> embeddedFile = file -> file.getFileName().toString().endsWith(".jpg") || file.getFileName().toString().endsWith(".jpeg")
+                || file.getFileName().toString().endsWith(".png") || file.getFileName().toString().endsWith(".gif") || file.getFileName().toString().endsWith(".svg")
+                || file.getFileName().toString().endsWith(".pdf");
```

Comment:
This could be a separate private method instead of a predicate.

Additionally, saving `file.getFileName().toString()` to a variable once would save a lot of duplication.